### PR TITLE
Contacts per case 0715

### PR DESCRIPTION
**Pull request by worldwidey**
- calculated number of contact per case from contact_records
- left_join case_records to contact_records in order to 1)insert 0 for the NA, 2)attached person_type
- added another color palette which has six colors since person_type now has 6 categories.
- drawed boxplot of the number of contacts per case.

### CODE REVIEW

File: uf_stp_report.Rmd [hunk 1]
```diff
@@ -104,6 +104,44 @@ contacts_who_became_cases_via_ufid_matching <- contact_records %>%
...
+                               TRUE~'Non-UF'))
+
+six_palette = c(
+    primary[1],
+    primary[2],
+    primary[3],
+    primary[4],
+    soft[4],
+    soft[5]
+    )
```
Comment (ChemiKyle) [inline uf_stp_report.Rmd:132]: This palette should be moved to `color_palettes.R`

File: uf_stp_report.Rmd [hunk 2]
```diff
@@ -104,6 +104,44 @@ contacts_who_became_cases_via_ufid_matching <- contact_records %>%
...
+  select(record_id,person_type,contact_num) %>% 
+  mutate_at(vars(contact_num),replace_na,'0') %>%
+  mutate_at("contact_num",as.numeric) %>%
+#  mutate_at(vars(person_type),replace_na,'Non-UF') %>% 
+  mutate(person_type=case_when(person_type=='EMPLOYEE'~'UF Employee',
+                               person_type=='STUDENT'~'UF Student',
+                               person_type=='BOTH'~'UF Employee and Student',
+                               person_type=='OTHER'~'Employee of UF Affiliate',
+                               person_type=='JAX'~'UF Employee or Student Located in Jacksonville',
+                               TRUE~'Non-UF'))
```
Comment (ChemiKyle) [inline uf_stp_report.Rmd:123]:
```suggestion
  mutate(person_type=case_when(person_type=='Employee'~'UF Employee',
                               person_type=='Student'~'UF Student',
                               person_type=='Both'~'UF Employee and Student',
                               person_type=='Other'~'Employee of UF Affiliate',
                               person_type=='JAX'~'UF Employee or Student Located in Jacksonville',
                               TRUE~'Non-UF'))
```

I changed the `person_type` in a different commit to get rid of the CAPS.

**Comment by ChemiKyle** [comment]
You built the plot but didn't show it anywhere, it should be rendered in the weekly data (and should use weekly data), use LaTeX to insert the figure, e.g.

https://github.com/ctsit/tt_reporting/blob/d2190f1cf96ecbe11b383e5b24692e54efdb7fd4/uf_stp_report.Rmd#L168-L175

**Comment by worldwidey** [comment]
> You built the plot but didn't show it anywhere, use LaTeX to insert the figure, e.g.
> 
> https://github.com/ctsit/tt_reporting/blob/d2190f1cf96ecbe11b383e5b24692e54efdb7fd4/uf_stp_report.Rmd#L168-L175

**Comment by ChemiKyle** [comment]
@worldwidey no need to close this PR, you can update it with more commits

File: uf_stp_report.Rmd (hunk 2 above)
Comment (worldwidey) [reply uf_stp_report.Rmd:123]: Isn't the data originally have Capital latter? This modification was for style?

File: uf_stp_report.Rmd [hunk 3]
```diff
@@ -104,6 +104,44 @@ contacts_who_became_cases_via_ufid_matching <- contact_records %>%
...
+    soft[4],
+    soft[5]
+    )
+
+boxplot_contact_per_case_all <- contact_per_case_all %>% 
+  mutate(person_type=fct_relevel(person_type,"UF Student","UF Employee","UF Employee and Student","Employee of UF Affiliate","UF Employee or Student Located in Jacksonville","Non-UF")) %>% 
+  ggplot(aes(x=person_type,y=contact_num, fill=person_type)) +
+  scale_fill_manual(values = six_palette) +
+  geom_boxplot() +
+  geom_jitter(width=0.1,alpha=0.2) +
```
Comment (ChemiKyle) [inline uf_stp_report.Rmd:139]:
```suggestion
  geom_jitter(width=0.25,height=0.25,alpha=0.2) +
```
The default jitter along the y-axis is 40% causing a bit of bleed within a `person_type`, `height=0.25` makes the grouping a bit tighter. I also upped the `width` to let the points spread out a bit more along the `person_type` bins.

Before:
image


After:
image

File: uf_stp_report.Rmd (hunk 2 above)
Comment (ChemiKyle) [reply uf_stp_report.Rmd:123]: When Laurence refactored in PR #17 it was all caps, before merging I added a small commit that converted `CAPS` to `Title` for the `persontype` dataframe to keep it in line with the appearance of other graphs (see comments in the linked PR)

**Pull request merged**